phpBB/phpbb/extension/base.php

```diff
@@ -137,6 +139,26 @@ protected function get_migration_file_list()
 
 		$migrations = $this->extension_finder->get_classes_from_files($migrations);
 
+		if ($validate_classes)
+		{
+			// Unset classes that do not exist or do not extend the
+			// abstract class phpbb\db\migration\migration
+			foreach ($migrations as $key => $migration)
+			{
+				if (class_exists($migration))
+				{
+					$reflector = new \ReflectionClass($migration);
```

Comment:
Based on my tests class_implements is about 35% faster (on average 0.9 µs for reflection class vs. 0.7 µs for class_implements). However, class_implements does not allow us to check if the class is instantiable.